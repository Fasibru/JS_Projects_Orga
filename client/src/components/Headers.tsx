import React from 'react';
import styled from 'styled-components';

const Headers: React.SFC = () => (
  <HeaderWrapper>
    <HeaderItem>Name</HeaderItem>
    <HeaderItem>Created</HeaderItem>
    <HeaderItem>Last modified</HeaderItem>
  </HeaderWrapper>
);

export const HeaderWrapper = styled.div`
  display: flex;
  font-weight: bold;
`;

export const HeaderItem = styled.div`
  flex: 1;
`;

export default Headers;
