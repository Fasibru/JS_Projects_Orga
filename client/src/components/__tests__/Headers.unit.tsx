import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Headers, { HeaderItem } from '../Headers';

describe('Test <Headers />', () => {
  let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

  beforeEach(() => {
    wrapper = shallow(<Headers />);
  });

  it('Snapshot is consistent', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Items are rendered correctly', () => {
    const title = wrapper.find(HeaderItem).at(0);
    const dateCreated = wrapper.find(HeaderItem).at(1);
    const dateModified = wrapper.find(HeaderItem).at(2);

    expect(title.text()).toEqual('Name');
    expect(dateCreated.text()).toEqual('Created');
    expect(dateModified.text()).toEqual('Last modified');
  });
});
