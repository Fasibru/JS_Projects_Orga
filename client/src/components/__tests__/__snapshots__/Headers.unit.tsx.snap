// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Headers /> Snapshot is consistent 1`] = `
<styled.div>
  <styled.div>
    Name
  </styled.div>
  <styled.div>
    Created
  </styled.div>
  <styled.div>
    Last modified
  </styled.div>
</styled.div>
`;
